tests/view_tests/tests/test_i18n.py

```diff
@@ -204,7 +204,7 @@ def test_setlang_decodes_http_referer_url(self):
         # The URL & view must exist for this to work as a regression test.
         self.assertEqual(reverse('with_parameter', kwargs={'parameter': 'x'}), '/test-setlang/x/')
         lang_code = self._get_inactive_language_code()
-        encoded_url = '/test-setlang/%C3%A4/'  # (%C3%A4 decodes to ä)
+        encoded_url = '/test-setlang/%C3%A4%26&/'  # (%C3%A4 decodes to ä, %26 to &)
```

Comment:
I feel the test could be slightly improved to better underline the problem that was reported and to prevent future regression. Something like:
```suggestion
        encoded_url = '/test-setlang/%C3%A4/?foo=bar&baz=alpha%26omega'  # (%C3%A4 decodes to ä, %26 to &)
```